bloom/generators/rosdebian.py

```diff
@@ -87,12 +87,27 @@ def fallback_resolver(key, peer_packages, rosdistro=self.rosdistro):
         )
         subs['Package'] = rosify_package_name(subs['Package'], self.rosdistro)
 
-        # XXX Add workspace package to runtime and buildtime dependencies for ROS 2 only.
-        if self.rosdistro in ['r2b2', 'r2b3', 'ardent', 'bouncy'] and \
-                package.name not in ['ament_cmake_core', 'ament_package', 'ros_workspace']:
-                    workspace_pkg_name = rosify_package_name('ros-workspace', self.rosdistro)
-                    subs['BuildDepends'].append(workspace_pkg_name)
-                    subs['Depends'].append(workspace_pkg_name)
+        # ROS 2 specific bloom extensions.
+        # TODO(nuclearsandwich) explore other ways to enable these extensions, reduce their necessity,
+        # or make them configurable rather than relying on distro names.
+        if self.rosdistro in ['r2b2', 'r2b3', 'ardent', 'bouncy']:
+            # Add ros-workspace package as a dependency to any package other
+            # than ros_workspace and its dependencies.
+            if package.name not in ['ament_cmake_core', 'ament_package', 'ros_workspace']:
+                workspace_pkg_name = rosify_package_name('ros-workspace', self.rosdistro)
+                subs['BuildDepends'].append(workspace_pkg_name)
+                subs['Depends'].append(workspace_pkg_name)
+
+            # Add vendor typesupport packages to build dependencies for rosidl_interface_packages.
+            if rosdistro in ['bouncy'] and 'rosidl_interface_packages' in package.member_of_groups:
+                ROS2_VENDOR_TYPESUPPORT_PACKAGES = [
+                        'rosidl_typesupport_connext_c',
+                        'rosidl_typesupport_connext_cpp',
+                        'rosidl_typesupport_opensplice_c',
+                        'rosidl_typesupport_opensplice_cpp',
+                        ]
```

Comment:
Nitpick: The indentation here is strange, I would have expected only one set of four spaces before the items and the closing bracket to align with the start of the variable name.